dds/DCPS/RcObject.h

```diff
@@ -31,10 +31,18 @@ namespace DCPS {
   class OpenDDS_Dcps_Export WeakObject : public PoolAllocationBase
   {
   public:
+
+#ifdef ACE_HAS_CPP11
+    typedef std::atomic<long> AtomicCountType;
+#define GET_ATOMIC_COUNT_VALUE(v) v
+#else
+    typedef ACE_Atomic_Op<ACE_SYNCH_MUTEX, long> AtomicCountType;
+#define GET_ATOMIC_COUNT_VALUE(v) v.value()
+#endif
```

Comment:
Is this worth moving to its own header? Could it be useful in other places?

Reply:
Possibly. I see we are using `int`, `long`, `unsigned long`, and `size_t` as atomic counts in other header files. Perhaps we could consolidate to a single header, if we could agree on the underlying integral type/types. Not sure which, if any, really need to be signed, but we could perhaps make one for `size_t` and `ssize_t` to be safe?

Reply:
But that effort feels like a good candidate for a separate task / PR if we decide to go that way.

Reply:
It could be a template parameter in that header

Reply:
If it's not going to be otherwise changed on this PR, rename the macro to start with OPENDDS.

Reply:
I've attempted to move the typedefs into their own header and opted to simply name them `Atomic`. I think I've removed the need for a macro through inheritance and addition of `operator T()` similar to `std::atomic`. Let me know if that's acceptable. I'm sure there are corner cases I'm not considering here, but for the most part I think it's sufficient for our needs for now. If you're happy with it, I can migrate other atomic counts to use it, or we can save that for a different task / PR.